pkg/api/v1/clients/kube/resource_client_factory.go

```diff
@@ -272,9 +311,36 @@ func (f *ResourceClientSharedInformerFactory) Start() {
 	})
 }
 
+// RegisterNewNamespace is used when the resource client is running. This will add a new namespace to the
+// kube controller so that events can be received.
+func (f *ResourceClientSharedInformerFactory) RegisterNewNamespace(namespace string, rc *ResourceClient) error {
+	// because this is an exposed function, Register New Namespace could be called at any time
+	// in that event, we want to make sure that the cache has started, the reason is because
+	// we have to initiallize the default namespaces as well as this new namespace
+	f.Start()
+	// we should only register a namespace once and only once
+	once, _ := f.registerNamespaceLock.LoadOrStore(namespace, &sync.Once{})
+	once.(*sync.Once).Do(func() {
+		ctx := f.ctx
+		if ctxWithTags, err := tag.New(ctx, tag.Insert(KeyKind, rc.resourceName)); err == nil {
+			ctx = ctxWithTags
+		}
+		informer, err := f.addNewNamespaceToRegistry(ctx, namespace, rc)
+		if err != nil {
+			// TODO-JAKE not sure if we want to panic here or to return an error.  But I feel like panics are ok.
+			contextutils.LoggerFrom(ctx).Panicf("failed to add new namespace to registry: %v", err)
+		}
+		if err := f.kubeController.AddNewInformer(informer); err != nil {
+			contextutils.LoggerFrom(ctx).Panicf("failed to add new informer to kube controller: %v", err)
+		}
+
+	})
```

Comment:
I think panicking here should not occur.  
The reason is because if the client adds a new namespace, then Gloo fails.